keycloak/openid_client.go

```diff
@@ -232,7 +232,9 @@ func (keycloakClient *KeycloakClient) getOpenidClientScopes(realmId, clientId, t
 	var scopes []*OpenidClientScope
 
 	err := keycloakClient.get(fmt.Sprintf("/realms/%s/clients/%s/%s-client-scopes", realmId, clientId, t), &scopes, nil)
-	if err != nil {
+	if err != nil && ErrorIs404(err) {
+		return nil, fmt.Errorf("validation error: client with id %s does not exist", clientId)
```

Comment:
was there a particular reason that this is needed?  normally, we check for 404 errors within the `provider` package.

Reply:
Good point. I saw a problem in the [TestAccKeycloakOpenidClientDefaultScopes_validateClientDoesNotExist](https://app.circleci.com/pipelines/github/mrparkers/terraform-provider-keycloak/1922/workflows/3c229825-8022-4926-9631-5e8e9acdd086/jobs/4184/tests#failed-test-0) test, and saw that the validation is missing when getting for scopes. Therefore, I added it.

It's strange that [this](https://github.com/mrparkers/terraform-provider-keycloak/blob/e0ead743558767528cd9a17e652673f3a1a10ec4/keycloak/openid_client.go#L273) check didn't work here.